### PR TITLE
[Do not merge][Datafactory] Add Tumbling Window dependsOn property to Trigger.json

### PR DESCRIPTION
**Pull request by alexsaff**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 1]
```diff
@@ -365,6 +372,24 @@
           "maximum": 86400
         }
       }
+    },
+    "TumblingWindowDependency": {
+      "description": "Tumbling Window dependency information.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Reference type \"TriggerReference\" for Tumbling Window.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Suggested description: "Tumbling window trigger reference type." This is consistent with our other descriptions of similar kind.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 2]
```diff
@@ -365,6 +372,24 @@
           "maximum": 86400
         }
       }
+    },
+    "TumblingWindowDependency": {
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Are we planning on having conditional dependency for triggers? If not then it is better to name this as "TumblingWindowTriggerReference" to be consistent with other naming conventions.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 3]
```diff
@@ -365,6 +372,24 @@
           "maximum": 86400
         }
       }
+    },
+    "TumblingWindowDependency": {
+      "description": "Tumbling Window dependency information.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Suggested description: "Tumbling window trigger reference type."

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 4]
```diff
@@ -365,6 +372,24 @@
...
         }
       }
+    },
+    "TumblingWindowDependency": {
+      "description": "Tumbling Window dependency information.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Reference type \"TriggerReference\" for Tumbling Window.",
+          "type": "string"
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: I think ARM folks will comment on this too, you should add an enum here:

"enum": [
    "TumblingWindowTriggerReference"
]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 5]
```diff
@@ -365,6 +372,24 @@
...
+        },
+        "referenceName": {
+          "description": "Trigger reference name.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: "Reference TumblingWindowTrigger name."

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 6]
```diff
@@ -324,6 +324,13 @@
             "retryPolicy": {
               "$ref": "#/definitions/RetryPolicy",
               "description": "Retry policy that will be applied for failed pipeline runs."
+            },
+            "dependsOn": {
+              "type": "array",
+              "description": "Tumbling Window depends on condition.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: In your case there is no condition, so suggested description: "Tumbling window triggers that this trigger depends on."

**Review by hvermis — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 6 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Resolved

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 3 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Resolved

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 1 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Resolved

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 5 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Resolved

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 4 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Resolved

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 2 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Resolved

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 7]
```diff
@@ -324,6 +324,13 @@
             "retryPolicy": {
               "$ref": "#/definitions/RetryPolicy",
               "description": "Retry policy that will be applied for failed pipeline runs."
+            },
+            "dependsOn": {
+              "type": "array",
+              "description": "Tumbling window triggers that this trigger depends on.",
+              "items": {
+                "$ref": "#/definitions/TumblingWindowDependency"
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: You need to change this too to "TumblingWindowTriggerReference"

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 8]
```diff
@@ -365,6 +372,27 @@
...
       }
+    },
+    "TumblingWindowTriggerReference": {
+      "description": "Tumbling window trigger reference type.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Tumbling window trigger reference type.",
+          "type": "string",
+          "enum": [
```
Comment (mcardosos) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: >enum [](start = 11, length = 4)

Also use [x-ms-enum](https://github.com/stankovski/AutoRest/blob/master/Documentation/swagger-extensions.md#x-ms-enum)

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 9]
```diff
@@ -365,6 +372,27 @@
           "maximum": 86400
         }
       }
+    },
+    "TumblingWindowTriggerReference": {
+      "description": "Tumbling window trigger reference type.",
+      "type": "object",
+      "properties": {
+        "type": {
```
[the remark this answers is not in the document]
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Here is a JSON example:
"dependsOn": [
        {
          "type": "TriggerReference",
          "referenceName": "mytumblingwindowtrigger1"
        },
        {
          "type": "TriggerReference",
          "referenceName": "mytumblingwindowtrigger2"
        }
      ],

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 8 above)
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Using x-ms-enum is blocked by hvermis

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (mcardosos) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: What I mean is, could the `type` have another value?

[the remark this answers is not in the document]
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: No, the type is not discriminator.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 8 above)
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: @mcardosos We don't want enum types generated for this, it is more of a hint to customer that this is the value. Here's another example where our swagger has the same structure: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json#L3024

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (alexsaff) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: At his point we are not planning "type" to have another value, may be in the future

**Comment by hvermis** [comment]
@mcardosos Please do not merge this yet as the backend is not ready yet.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (mcardosos) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: okok

**Comment by mcardosos** [comment]
@hvermis When will this be available?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json (hunk 8 above)
Comment (mcardosos) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: OKok

**Review by mcardosos — APPROVED** [review]
For awareness, this swagger has [linter errors](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/368722141#L744). I know they were not introduced in this PR

**Comment by salameer** [comment]
@hvermis please ping this thread when your ready to merge this swagger

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 10]
```diff
@@ -365,6 +372,27 @@
...
+      "properties": {
+        "type": {
+          "description": "Tumbling window trigger reference type.",
+          "type": "string",
+          "enum": [
+            "TriggerReference"
+          ]
+        },
+        "referenceName": {
+          "description": "Reference TumblingWindowTrigger name.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Update the description too

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 11]
```diff
@@ -365,6 +372,27 @@
           "maximum": 86400
         }
       }
+    },
+    "TumblingWindowTriggerReference": {
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Change the type name here too

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 12]
```diff
@@ -324,6 +324,13 @@
             "retryPolicy": {
               "$ref": "#/definitions/RetryPolicy",
               "description": "Retry policy that will be applied for failed pipeline runs."
+            },
+            "dependsOn": {
+              "type": "array",
+              "description": "Tumbling window triggers that this trigger depends on.",
+              "items": {
+                "$ref": "#/definitions/TumblingWindowTriggerReference"
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Change it here too

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 13]
```diff
@@ -324,6 +324,13 @@
             "retryPolicy": {
               "$ref": "#/definitions/RetryPolicy",
               "description": "Retry policy that will be applied for failed pipeline runs."
+            },
+            "dependsOn": {
+              "type": "array",
+              "description": "Tumbling window triggers that this trigger depends on.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Leave the description here the same to specifically indicate that it is a tumbling window trigger

Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: You can also add - "Only tumbling window triggers are supported." If we ever add other trigger types here we will update the description.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 14]
```diff
@@ -365,6 +372,27 @@
           "maximum": 86400
         }
       }
+    },
+    "TriggerReference": {
+      "description": "Tumbling window trigger reference type.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Remove "Tumbling window"  from description.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json [hunk 15]
```diff
@@ -365,6 +372,27 @@
           "maximum": 86400
         }
       }
+    },
+    "TriggerReference": {
+      "description": "Tumbling window trigger reference type.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Tumbling window trigger reference type.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json]: Remove "Tumbling window"  from description.

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by hvermis** [comment]
@alexsaff I beleive this PR should be closed and new one opened by Nalini?

**Comment by alexsaff** [comment]
This PR is abandoned and will be replaced by a new one with a different data model

**Pull request closed without merging**